Frameworks/Foundation/NSKeyValueCoding.mm

```diff
@@ -185,13 +175,12 @@ bool KVCGetViaAccessor(NSObject* self, SEL getter, id* ret) {
         return false;
     }
 
-    if (0
-#define APPLY_TYPE(type, name, capitalizedName, encodingChar) || (valueType[0] == encodingChar && (*ret = quickGet<type>(self, getter)))
-        APPLY_TYPE(id, object, Object, '@') APPLY_TYPE(Class, class, Class, '#')
+    switch (valueType[0]) {
+#define APPLY_TYPE(type, name, capitalizedName, encodingChar) case encodingChar: *ret = quickGet<type>(self, getter); return true;
+        APPLY_TYPE(id, object, Object, '@')
+        APPLY_TYPE(Class, class, Class, '#')
 #include "type_encoding_cases.h"
 #undef APPLY_TYPE
```

Comment:
well this is mildly terrifying. I'm pretty positive we shouldn't be knowingly including a .h smack dab in the middle of a function. That is asking for trouble and abusing the preprocessor in a way I'm not ok with. Having that header define macros etc to make this easier would be acceptable but not this.